modules/core/src/smithy4s/ShapeId.scala

```diff
@@ -28,8 +28,13 @@ object ShapeId extends ShapeTag.Has[ShapeId] {
     if (!string.contains('#')) None
     else {
       val segments = string.split("#")
-      if (segments.length > 1) None
-      else Some(ShapeId(segments(0), segments(1)))
+      if (
+        segments.length == 2 &&
+        segments(0).nonEmpty &&
+        segments(1).nonEmpty
+      )
+        Some(ShapeId(segments(0), segments(1)))
+      else None
```

Comment:
**thought:** It's still a pretty simplistic implementation, long-term we might want to do something like https://github.com/disneystreaming/smithy-translate/blob/659c266a1c301440fcefdb1b90467fb0e12e3bea/modules/formatter/src/parsers/ShapeIdParser.scala, but for now I just wanted to address the immediate issue - valid shapes not parsing.